config.py

```diff
@@ -226,62 +178,112 @@ def __with_lock_and_backup(self, path: Path, backup: bool, callable: Callable[[]
                     raise Exception(f"Failed to save config: {e}")
         finally:
             # Cleanup lock and backup files:
-            if lock_path.exists():  # Only try to delete if it exists
-                try:
-                    lock_path.unlink()
-                except FileNotFoundError:
-                    pass  # Ignore if file doesn't exist
-            
+            if lock_path.exists():
+                lock_path.unlink()
+
             if backup and backup_path and backup_path.exists():
-                try:
-                    backup_path.unlink()
-                except FileNotFoundError:
-                    pass  # Ignore if file doesn't exist
+                backup_path.unlink()
 
     @abstractmethod
-    def _handle_loaded_config(self, data):
+    def _handle_loaded_config(self, json_string: str):
         pass
 
-class AlltalkNewEnginesConfig(AbstractJsonConfig):
+    def to_dict(self):
+        return self.__delegate.model_dump(by_alias=True)
+
+    # Delegation to the loaded config data
+    @property
+    def _xtra(self): return [o for o in dir(self.__delegate) if not o.startswith('_')]
+
+    def __getattr__(self, key):
+        if key in self._xtra: return getattr(self.__delegate, key)
+        raise AttributeError(key)
+
+    def __getattribute__(self, key):
+        if key.startswith("_"):
+            # Private and protected attributes always are on self:
+            return super().__getattribute__(key)
+        if key in self._xtra:
+            # Delegate attribute
+            return getattr(self.__delegate, key)
+        # Get own attribute:
+        return super().__getattribute__(key)
+
+    def __setattr__(self, key, value):
+        if key.startswith("_"):
+            # Private and protected attributes always are on self:
+            super().__setattr__(key, value)
+        elif key in dir(self.__delegate):
+            # Delegate attribute
+            setattr(self.__delegate, key, value)
+        else:
+            # Set own attribute:
+            super().__setattr__(key, value)
+
+    def __dir__(self):
+        def custom_dir(c, add): return dir(type(c)) + list(c.__dict__.keys()) + add
+        return custom_dir(self, self._xtra)
+
+class AlltalkNewEnginesConfigFields:
+    engines_available: MutableSequence[AlltalkAvailableEngine] = Field(default_factory=list)
+
+class AlltalkNewEnginesConfigModel(BaseModel, AlltalkNewEnginesConfigFields):
+
+    def get_engine_names_available(self):
+        return [engine.name for engine in self.engines_available]
+
+    def get_engines_matching(self, condition: Callable[[AlltalkAvailableEngine], bool]):
+        return [x for x in self.engines_available if condition(x)]
+
+class AlltalkNewEnginesConfig(AbstractJsonConfig, AlltalkNewEnginesConfigFields):

```

Comment:
There are more classes involved now. Let me explain why:

- `AlltalkNewEnginesConfig` is the class we already had before. It is what is used in code to access configuration properties. However, the class does not directly contain the properties, but delegates to `AlltalkNewEnginesConfigModel`.
- `AlltalkNewEnginesConfigModel` is a pydantic model (`BaseModel`), delaing with deserialization and serialization.
- `AlltalkNewEnginesConfigFields` contains the actual properties of a config.

Now you might wonder why `AlltalkNewEnginesConfig` and `AlltalkNewEnginesConfigModel` both inherit from `AlltalkNewEnginesConfigFields`. This is because my IDE did not recognize the delegated properties in `AlltalkNewEnginesConfig`, so no tab completion. The inheritances fixes it, even though properties are actually delegated.